File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string txt);
int count_words(string txt);
int count_sentances(string txt);


int main(void)
{
    string text = get_string("Text:");

    int letter_count = count_letters(text);
    int word_count = count_words(text);
    int sentence_count = count_sentances(text);

    printf("%i %i %i \n", letter_count, word_count, sentence_count);

}

int count_letters(string txt)
{
    int count = 0;

    for(int i = 0; i < strlen(txt); i++)
    {
        if ((txt[i] >= 97 && txt[i] <= 122) || (txt[i] >= 65 && txt[i] <= 90))
        {
            count ++;
        }
    }

    return(count);
}

int count_words(string txt)
{
    int count = 0;

    for(int i = 0; i < strlen(txt); i++)
    {
        if (!(txt[i] >= 97 && txt[i] <= 122) && !(txt[i] >= 65 && txt[i] <= 90) && !(txt[i] == 96)
            ((txt[i-1] >= 97 && txt[i-1] <= 122) || (txt[i-1] >= 65 && txt[i-1] <= 90)))
        {
            count ++;
        }
    }

    return(count);
}

int count_sentances(string txt)
{
    int count = 0;

    for(int i = 0; i < strlen(txt); i++)
    {
        if (txt[i] == 33 || txt[i] == 46 || txt[i] == 63 )
        {
            count ++;
        }
    }

    return(count);
}

File: sentimental-readability/readability.py
def main():

    text = str(input("Text: "))

    sentances = count_sentances(text)
    words = count_words(text)
    letters = count_letters(text)

    print(f"{sentances}--{words}--{letters}")

    L = 0
    S = 0

    grade = 0.0588 * L - 0.296 * S - 15.8


def count_sentances(txt:str):
    count = 0

    for i in range(len(txt)):
        if txt[i] in [".","?","!"] and txt[i-1] not in [".","?","!"]:
            count += 1
    return count


def count_words(txt:str):
    count = 0

    for i in range(len(txt)):
        if not txt[i].isalnum() and txt[i-1].isalnum():
            count += 1
    return count


def count_letters(txt:str):
    count = 0

    for i in range(len(txt)):
        if txt[i].isalnum():
            count += 1
    return count


if __name__ == "__main__":
    main()
